### ISSUE TITLE
Un-handled .NET error in shared workbooks.

### ISSUE THREAD

**Issue by ptwales** [retailcoder/Rubberduck#365]
An un-handled error is raised when opening the VBE of a shared macro enabled workbook or trying to use Rubberduck's features.  Shared workbooks cannot have their VBA code read or written, which is causing the error.  For the same reason we don't need Rubberduck to work in a shared workbook.  However, this case should be handled more gracefully than the scary .NET error.  Either just exit silently or raise a "You can't use Rubberduck in a shared workbook error".

Steps to reproduce with Rubberduck installed:

1. Create a Workbook
1. Add a Code Module to the VB Project
1. Save as Macro Enabled Workbook (or binary)
1. Share Workbook
1. Close and re-open workbook
1. Open VBE.
1. Receive Unhandled Error _Can't Perform operation since the project is protected_. Details are at the bottom.
  2. Opening TODO items or Code Inspections will re-raise the error.

```
    See the end of this message for details on invoking 
    just-in-time (JIT) debugging instead of this dialog box.

    ************** Exception Text **************
    System.Runtime.InteropServices.COMException (0x800AC471): Can't perform operation since the project is protected.
       at Microsoft.Vbe.Interop._VBProject.get_VBComponents()
       at Rubberduck.UI.CodeExplorer.CodeExplorerDockablePresenter.<AddModuleNodesAsync>d__15.MoveNext() in c:\Users\Christopher\Source\Repos\Rubberduck\RetailCoder.VBE\UI\CodeExplorer\CodeExplorerDockablePresenter.cs:line 107
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Rubberduck.UI.CodeExplorer.CodeExplorerDockablePresenter.<>c__DisplayClassd.<<AddProjectNodes>b__c>d__f.MoveNext() in c:\Users\Christopher\Source\Repos\Rubberduck\RetailCoder.VBE\UI\CodeExplorer\CodeExplorerDockablePresenter.cs:line 97
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<ThrowAsync>b__4(Object state)


    ************** Loaded Assemblies **************
    mscorlib
        Assembly Version: 4.0.0.0
        Win32 Version: 4.0.30319.34209 built by: FX452RTMGDR
        CodeBase: file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll
    ----------------------------------------
    Rubberduck
        Assembly Version: 1.2.5541.25883
        Win32 Version: 1.2.0.3
        CodeBase: file:///C:/Program%20Files/Rubberduck/Rubberduck/Rubberduck.dll
    ----------------------------------------
    System.Windows.Forms
        Assembly Version: 4.0.0.0
        Win32 Version: 4.0.30319.34209 built by: FX452RTMGDR
        CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
    ----------------------------------------
    System.Drawing
        Assembly Version: 4.0.0.0
        Win32 Version: 4.0.30319.34209 built by: FX452RTMGDR
        CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
    ----------------------------------------
    System
        Assembly Version: 4.0.0.0
        Win32 Version: 4.0.30319.34238 built by: FX452RTMGDR
        CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
    ----------------------------------------
    System.Core
        Assembly Version: 4.0.0.0
        Win32 Version: 4.0.30319.34209 built by: FX452RTMGDR
        CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
    ----------------------------------------
    Antlr4.Runtime.net45
        Assembly Version: 4.3.0.0
        Win32 Version: 4.3.0.0
        CodeBase: file:///C:/Program%20Files/Rubberduck/Rubberduck/Antlr4.Runtime.net45.DLL
    ----------------------------------------
    System.Xml
        Assembly Version: 4.0.0.0
        Win32 Version: 4.0.30319.34234 built by: FX452RTMGDR
        CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
    ----------------------------------------
    System.Configuration
        Assembly Version: 4.0.0.0
        Win32 Version: 4.0.30319.34209 built by: FX452RTMGDR
        CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
    ----------------------------------------
    Microsoft.GeneratedCode
        Assembly Version: 1.0.0.0
        Win32 Version: 4.0.30319.34234 built by: FX452RTMGDR
        CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
    ----------------------------------------
    Accessibility
        Assembly Version: 4.0.0.0
        Win32 Version: 4.0.30319.34209 built by: FX452RTMGDR
        CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/Accessibility/v4.0_4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll
    ----------------------------------------
    CustomMarshalers
        Assembly Version: 4.0.0.0
        Win32 Version: 4.0.30319.34209 built by: FX452RTMGDR
        CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_32/CustomMarshalers/v4.0_4.0.0.0__b03f5f7f11d50a3a/CustomMarshalers.dll
    ----------------------------------------

    ************** JIT Debugging **************
    To enable just-in-time (JIT) debugging, the .config file for this
    application or computer (machine.config) must have the
    jitDebugging value set in the system.windows.forms section.
    The application must also be compiled with debugging
    enabled.

    For example:

    <configuration>
        <system.windows.forms jitDebugging="true" />
    </configuration>

    When JIT debugging is enabled, any unhandled exception
    will be sent to the JIT debugger registered on the computer
    rather than be handled by this dialog box.
```

**Comment by ckuhn203** [comment]
If I'm reading the stack trace right, you have Version 1.2 or 1.21 installed. 
I believe this issue was corrected in 1.22 

https://github.com/retailcoder/Rubberduck/releases/tag/v1.22

**Comment by ptwales** [comment]
Updating to 1.22 yields no error.  Issue was already solved.
